tests/gateway/test-nodeport.yaml

```diff
@@ -0,0 +1,20 @@
+config:
+  serverConfig:
+    http-server.http.enabled: true
+    http-server.http.port: 8080
+    http-server.https.enabled: true
+    http-server.https.port: 8443
+    http-server.https.keystore.path: /etc/scratch/tls.pem
+
+service:
+  spec:
```

Comment:
We're not reading any properties from this `spec` key, so if this test passes without these settings, it means it's not a good test. Should it be using different port numbers?

Reply:
I outsmarted myself with this logic in `charts/gateway/templates/test-nodeport.yaml`:

```
        - |
          {{- if eq .Values.service.type "NodePort" }}
          [ "$(wget --no-check-certificate https://${NODE_IP}:{{ index .Values.service.ports 0 "nodePort"}}/entity/GATEWAY_BACKEND -O -)" = "[]" ] &&
          [ "$(wget http://${NODE_IP}:{{ index .Values.service.ports 1 "nodePort"}}/entity/GATEWAY_BACKEND -O -)" = "[]" ]
          {{- else }}
          echo non NodePort service type
          {{- end }}
```

the invalid key effectively disabled the test. I think this can be fixed by using `serverConfig` as the independent variable

Reply:
The values files using `service.spec` failed after the correction to `connection-test.yaml` to check if `serverConfig.http-server.http[s].enabled = =true`, so I think the test is now good